### PR TITLE
Prefer pytest-style tests (issue #861)

### PR DESCRIPTION
**Pull request by mfonism**
This PR converts xunit-style style tests to pytest-style tests as per issue #861

### CODE REVIEW

**Comment by mfonism** [comment]
@bsipocz @ceb8 Can you please add a milestone to this so I can add a changelog for the work done.

**Comment by bsipocz** [comment]
Thank you @mfonism. Overall looks good, but this will need careful review which would take a bit of a time.

**Review by ceb8 — APPROVED** [review]
Looks good to me. I had one suggestion about cleaning up strings but all the functionality looks good.

File: astroquery/utils/tap/conn/tests/test_conn.py [hunk 1]
```diff
@@ -26,132 +25,99 @@ def data_path(filename):
...
+    # TapConn
+    tap = TapConn(ishttps=False,
+                  host=host,
+                  server_context=serverContext,
+                  tap_context=tapContext,
+                  port=connPort,
+                  sslport=connPortSsl,
+                  connhandler=conn)
+    hostUrl = host + ":" + str(connPort) + "/" + serverContext + "/" \
+        + tapContext + "/"
```
[the remark this answers is not in the document]
Comment (bsipocz) [reply astroquery/utils/tap/conn/tests/test_conn.py:45]: I would even suggest to use fstrings, those are even easier to read.

[the remark this answers is not in the document]
Comment (ceb8) [reply astroquery/utils/tap/conn/tests/test_conn.py:45]: Oooh, yes! I forgot our earliest supported python version has those.
F-strings are the best: `f"{host}:{connPort}/{serverContext}/{tapContext}"`

[the remark this answers is not in the document]
Comment (mfonism) [reply astroquery/utils/tap/conn/tests/test_conn.py]: Done in 2bc3e44

**Review by bsipocz — APPROVED** [review]
I've reviewed the last commit about the strings, and will use the previous approval for the test refactor for merging.

**Comment by bsipocz** [comment]
Oh, wow, you changed all the strings 🎉

**Comment by bsipocz** [comment]
So, this is a massive diff, I gave a review for the string changes and didn't spot any issues. And @ceb8  has already approved the test refactors. So this is ready to go now.
(I've also confirmed that the remote tests in all affected modules still pass, once rebased to get the recent changes to esasky)

**Comment by bsipocz** [comment]
Massive thank you @mfonism!

**Comment by mfonism** [comment]
Thank you @bsipocz and @ceb8. It's a pleasure getting such warm and kind feedback from you.

**Pull request merged**